rules/xcodeproj.bzl

```diff
@@ -147,10 +151,16 @@ def _xcodeproj_aspect_impl(target, ctx):
                     hmap_paths = depset(hmap_paths),
                 ),
             )
-        direct_targets = [info]
+
+        direct_targets = []
+        transitive_targets = []
+        if not _IGNORE_AS_TARGET_TAG in tags:
+            direct_targets.append(info)
+            transitive_targets.append(info)
+
         if test_host_target:
             direct_targets.extend(test_host_target[_TargetInfo].direct_targets)
-        target_info = _TargetInfo(direct_targets = direct_targets, targets = depset([info], transitive = _get_attr_values_for_name(deps, _TargetInfo, "targets")))
+        target_info = _TargetInfo(direct_targets = direct_targets, targets = depset(transitive_targets, transitive = _get_attr_values_for_name(deps, _TargetInfo, "targets")))
```

Comment:
@segiddins are above what you have in mind? I have to do the same to `transitive_targets ` because the sanity check I added fails if `include_transittive_targets` is set to True, I am adding a new xcodeproj rule for Test project where it is set to True but it fails current checks on hmap and framework search path. I need to figure out what's going on then

Reply:
Yes, this is roughly what I had in mind